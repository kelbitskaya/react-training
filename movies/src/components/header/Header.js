import React from 'react';
import Logo from '../common/Logo';
import Button from '../common/button/Button';

export default function Header() {
  return (
    <header>
      <div>
        <div className="header-line">
          <Logo />
          <Button className="button" title="+ Add movie" />
        </div>
        <div className="movie-search">
          <h1>find your movie</h1>
          <div className="movie-search__line">
            <input type="text" name="movie" className="movie-search__input" placeholder="What do you want to watch?" />
            <Button className="button button_search" title="Search" />
          </div>
        </div>
      </div>
    </header>
  );
}
