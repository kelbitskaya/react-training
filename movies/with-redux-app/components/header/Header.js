import React, {useEffect} from 'react';
import {connect, useDispatch} from 'react-redux';
import PropTypes from 'prop-types';
import Logo from '../common/Logo';
import Search from '../search/Search';
import AddButton from '../addButton/AddButton';
import MovieCard from '../movieCard/MovieCard';
import SearchButton from '../searchButton/SearchButton';
import {getMovieById} from '../../store/actions/actions';
import { useRouter } from 'next/router';


const mapStateToProps = (state) => {
  return {
    movie: state.currentMovie,
  }
};

const matchDispatchToProps = {getMovie: getMovieById};

const setMovieGenre = (movie) => movie.genres && movie.genres.join(', ');
const setMovieYear = (movie) => movie.release_date && +movie.release_date.substr(0, 4);


const Header = ({movie}) => {

  const router = useRouter();
  console.log(router);


  const dispatch = useDispatch();
  const { pathname } = router;

  const redirectToHomePage = () => {
    router.push('/');
  };


  useEffect(() => {
    if (router.asPath.includes('film')) {
      const movieId = router.asPath.slice(6);
      dispatch(getMovieById(movieId));
    }
  }, [router.asPath]);



  return (
    <header className="header">
      <div className="header__content">
        <div className="header-line">
          <Logo />
        </div>

        {
          movie ?
            <>
              <SearchButton/>
              <MovieCard
                title={movie.title}
                genre={setMovieGenre(movie)}
                year={movie.year}
                releaseDate={setMovieYear(movie)}
                src={movie.poster_path}
                url={movie.poster_path}
                id={movie.id}
                key={movie.id}
                overview={movie.overview}
                runtime={movie.runtime}
                description={movie.overview}
                rating={movie.vote_average}
              />
            </>
            :
            <>
              <AddButton/>
              <div className="movie-search">
                <h1 className="header__title">find your movie</h1>
                <Search
                  placeholder="What do you want to watch?"
                />
              </div>
            </>
        }

      </div>

    </header>
  );
};

Header.propTypes = {
  movie: PropTypes.shape({
    length: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    poster_path: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired,
    vote_average: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    overview: PropTypes.string.isRequired,
    runtime: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    genre: PropTypes.arrayOf(PropTypes.string).isRequired,
    selectMovie: PropTypes.func.isRequired,
    map: PropTypes.func.isRequired,
  }),
};

Header.defaultProps = {
  movie: {},
};

export default connect(mapStateToProps, matchDispatchToProps)(Header);
