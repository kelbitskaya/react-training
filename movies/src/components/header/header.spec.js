import React from 'react';
import Header from './Header';
import { Provider } from "react-redux";
import { createMockStore } from 'redux-test-utils';
import rerender from "react-test-renderer";
import routeData from "react-router";
import { BrowserRouter as Router } from 'react-router-dom';

let store = createMockStore('');
const resetStore = () => store = createMockStore({
  movie: {
    length: 11
  }
});

const mockLocation = {
  pathname: '/welcome',
  hash: '',
  search: '',
  state: ''
};

const props = {
  movie1: {
    length: 1,
    title: "",
    src: "",
    id: "",
    poster_path: "",
    releaseDate: "",
    vote_average: "",
    url: "",
    overview: "",
    runtime: "",
    description: "",
    year: 1,
    rating: 1,
    genre: [],
    selectMovie: jest.fn(),
    map: jest.fn(),
    isSubmitting: jest.fn(),
  },
  movie: {
    length: 11
  }
};

describe("should render Header component", ()=>{
  it("Header", ()=> {
    resetStore();
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
    const component = rerender.create(
      <Provider store={store}>
        <Router>
          <Header {...props}/>
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
