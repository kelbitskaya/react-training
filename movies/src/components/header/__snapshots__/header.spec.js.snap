// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component Header 1`] = `
<header
  className="header"
>
  <div
    className="header__content"
  >
    <div
      className="header-line"
    >
      <a
        aria-current="page"
        className="logo active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <strong
          className="logo__text"
        >
          netflix
        </strong>
        roulette
      </a>
    </div>
    <div
      className="App"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        + Add movie
      </button>
      <div />
    </div>
    <div
      className="movie-search"
    >
      <h1
        className="header__title"
      >
        find your movie
      </h1>
      <form
        className="movie-search__line"
        onSubmit={[Function]}
      >
        <input
          className="movie-search__input"
          id="title"
          onChange={[Function]}
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <button
          className="button button_search"
          onClick={[Function]}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</header>
`;

exports[`should render noMovie component should contain .logo-wrap wrapper 1`] = `
<header
  className="header"
>
  <div
    className="header__content"
  >
    <div
      className="header-line"
    >
      <a
        aria-current="page"
        className="logo active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <strong
          className="logo__text"
        >
          netflix
        </strong>
        roulette
      </a>
    </div>
    <div
      className="App"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        + Add movie
      </button>
      <div />
    </div>
    <div
      className="movie-search"
    >
      <h1
        className="header__title"
      >
        find your movie
      </h1>
      <form
        className="movie-search__line"
        onSubmit={[Function]}
      >
        <input
          className="movie-search__input"
          id="title"
          onChange={[Function]}
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <button
          className="button button_search"
          onClick={[Function]}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</header>
`;
